subprojects/workers/src/main/java/org/gradle/process/internal/daemon/WorkerDaemonClientsManager.java

```diff
@@ -79,4 +80,20 @@ public void stop() {
             allClients.clear();
         }
     }
+
+    /**
+     * Select idle daemon clients to stop.
+     *
+     * @param selectionFunction Gets all idle daemon clients, daemons of returned clients are stopped
+     */
+    public void selectIdleClientsToStop(Transformer<List<WorkerDaemonClient>, List<WorkerDaemonClient>> selectionFunction) {
+        synchronized (lock) {
+            List<WorkerDaemonClient> clientsToStop = selectionFunction.transform(new ArrayList<WorkerDaemonClient>(idleClients));
```

Comment:
Should we attempt to sort the idle clients (either by last use time or frequency of uses) so that we stop the most appropriate clients first?  Right now it looks like we just go through them in order until we've freed up enough memory.